students/eldonab/lesson04/mailroom2.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+from pathlib import Path
+
+#create a list of at least 5 donors.
+#updated donors_list from part one using a dictionary
+donors_list = {
+"Bill Gates": [50000, 600000],
+"Jeff Bezos": [100000, 400, 7000],
+"Hannah Smith": [400000, 600000], 
+"John Clark": [30000, 40000],
+"Andrew Jones": [80000,90000, 1000]
+}
+
+#updated all functions using dictionary methods:
+
+def add_donation_to_list(res_name):
+    """Add a new donation amount to an existing donor's name."""
+    amount = float(input("How much is the new donation amount? "))
+    for donor in donors_list:
+        if res_name in donor:
+            donors_list[donor].append(amount)
+            letter(res_name, amount)
+
+
+def show_list():
+    """Show existing list of donor names"""
+    for donor in donors_list:
+        print(donor)
+
+
+
+
+def add_new_donor(new_donor):
+    """Add a new donor name and donation amount."""
+    amount = float(input("How much is the new donation amount? "))
+    donors_list.update({new_donor: [amount]})
+    letter(new_donor, amount)
+   
+
+
+def is_in_list(donor_name):
+    """Check whether donor already exists in the list of donors."""
+    for donor in donors_list:
+        if donor_name in donor:
+            return True
+
+
+def letter(donor_name, donor_amount):
+    """Print a thank you letter"""
+    print("Dear {0}, thank you for your generous donation in the amount of ${1}!".format(donor_name, donor_amount))
+
+
+
+
+def thank_you_note():
```

Comment:
too many spaces, your module functions should have 2 lines space between and comments for function as docstrings or inline within a function.